src/components/Publish/_validation.ts

```diff
@@ -1,13 +1,58 @@
+import { FileInfo } from '@oceanprotocol/lib'
 import { MAX_DECIMALS } from '@utils/constants'
-import * as Yup from 'yup'
 import { getMaxDecimalsValidation } from '@utils/numbers'
-import { FileInfo } from '@oceanprotocol/lib'
+import * as Yup from 'yup'
+import { SchemaLike } from 'yup/lib/types'
 import { testLinks } from '../../@utils/yup'
+import { paramTypes } from '../@shared/FormInput/InputElement/ConsumerParameters'
+import { AlgorithmConsumerParameter } from './_types'
 
 // TODO: conditional validation
 // e.g. when algo is selected, Docker image is required
 // hint, hint: https://github.com/jquense/yup#mixedwhenkeys-string--arraystring-builder-object--value-schema-schema-schema
 
+const validationConsumerParameters: {
```

Comment:
suggestion: reuse the consumerParameter validation from publish/edit
we can think about moving the `validationConsumerParameter` const into the `InputElement/ConsumerParameters` directory and just import it here and in edit form validation.